app/controllers/user.js

```diff
@@ -171,11 +171,18 @@ module.exports = {
   updatePassword: async (req, res, next) => {
     const { password, id } = req.body
     const { token } = req.params
+    let preChange = true;
     try {
-      const decodedToken = jwt.verify(token, process.env.JWT_SECRET)
-
+      const user = await User.findById(id)
+      const decodedToken = jwt.verify(token, user.password, (err, token) => {
+        if (err){
+          preChange = false;
```

Comment:
Instead of using this preChange, throw the error directly since everything is in try block.